src/web-push-lib.js

```diff
@@ -55,6 +55,36 @@ WebPushLib.prototype.setVapidDetails =
       return;
     }
 
+    if (typeof subject !== 'string' || subject.length === 0) {
```

Comment:
Perhaps we can move the checks in a separate function in vapid-helper.js and use it everywhere, so we don't have duplicate code?